sagemaker-pipeline-compare-model-versions/notebook.ipynb

```diff
@@ -0,0 +1,703 @@
+    "\n",
+    "Run the pipeline again, but with a larger dataset. \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Execute pipeline with explicit parameters\n",
+    "execution = pipeline.start(\n",
+    "    parameters=dict(\n",
+    "        InputData=large_input_data_uri,\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "execution.wait()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Visualize\n",
+    "Select both versions and right-click. Choose `Compare model versions`.\n",
+    "\n",
+    "\n",
+    "![Compare versions](images/compare-versions.png \"Compare versions\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Clean up (optional)\n",
+    "Delete the model registry and the pipeline to keep the studio environment tidy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def delete_model_package_group(sm_client, package_group_name):\n",
+    "    try:\n",
+    "        model_versions = sm_client.list_model_packages(ModelPackageGroupName=package_group_name)\n",
+    "\n",
+    "    except Exception as e:\n",
+    "        print(\"{} \\n\".format(e))\n",
+    "        return\n",
+    "\n",
+    "    for model_version in model_versions[\"ModelPackageSummaryList\"]:\n",
+    "        try:\n",
+    "            sm_client.delete_model_package(ModelPackageName=model_version[\"ModelPackageArn\"])\n",
+    "        except Exception as e:\n",
+    "            print(\"{} \\n\".format(e))\n",
+    "        time.sleep(0.5)  # Ensure requests aren't throttled\n",
+    "\n",
+    "    try:\n",
+    "        sm_client.delete_model_package_group(ModelPackageGroupName=package_group_name)\n",
+    "        print(\"{} model package group deleted\".format(package_group_name))\n",
+    "    except Exception as e:\n",
+    "        print(\"{} \\n\".format(e))\n",
+    "    return\n",
+    "\n",
+    "\n",
+    "def delete_sagemaker_pipeline(sm_client, pipeline_name):\n",
+    "    try:\n",
+    "        sm_client.delete_pipeline(\n",
+    "            PipelineName=pipeline_name,\n",
+    "        )\n",
+    "        print(\"{} pipeline deleted\".format(pipeline_name))\n",
+    "    except Exception as e:\n",
+    "        print(\"{} \\n\".format(e))\n",
+    "        return"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import time\n",
+    "\n",
+    "client = boto3.client(\"sagemaker\")\n",
+    "\n",
+    "delete_model_package_group(client, model_package_group_name)\n",
+    "delete_sagemaker_pipeline(client, pipeline_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
```

Comment:
this is bit annoying feature of jupyter to carry the instance type and kernel type info in metadata. its better to remove this. and mention at the top of the notebook that this notebooks runs on Python3(DataScience) kernel

Reply:
Added a note about which kernel is used in the disclaimer, but not familiar with how I can remove the metadata.